src/main/kotlin/org/rust/lang/core/psi/RsCodeFragment.kt

```diff
@@ -162,4 +162,5 @@ class RsReplCodeFragment(fileViewProvider: FileViewProvider, override var contex
     val tailExpr: RsExpr?
```

Comment:
I'd suggest go the `org.rust.lang.core.psi.ext.RsBlockKt#getExpandedStmtsAndTailExpr` way here. It can be done even without macro expansion usage (but would be nice to use expanded macros in REPL someday), like this:
```Kotlin
val expandedStmtsAndTailExpr: Pair<List<RsExpandedElement>, RsExpr?>
    get() {
        val expandedElements = childrenOfType<RsExpandedElement>()
        val tailExpr = expandedElements.lastOrNull()?.let { it as? RsExpr }
        return when (tailExpr) {
            null -> expandedElements
            else -> expandedElements.subList(0, expandedElements.size - 1)
        } to tailExpr
    }
```
Then you can use `getExpandedStmtsAndTailExpr` instead of `stmts`, `tailExpr`, and `items` in type inference/name resolution and `org.rust.ide.console.RsConsoleOneCommandContext`.

Or maybe you will come up with another way of doing this, it'd be OK too.